lightning/src/ln/channelmanager.rs

```diff
@@ -291,6 +291,32 @@ struct ReceiveError {
 	msg: &'static str,
 }
 
+/// This enum is used to specify which error data to send to peers when failing back an HTLC
+/// using [`ChannelManager::fail_htlc_backwards_with_reason`].
+///
+/// For more info on failure codes, see <https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages>.
+#[derive(Clone, Copy)]
+pub enum FailureCode {
+	/// A general temporary failure for otherwise unspecified errors that occur
+	/// for the entire node while handling the received onion. This can be used in situations
+	/// where other temporary error codes would not fit, while indicating to the origin node
+	/// it may want to retry the payment.
```

Comment:
I think this could be a bit more concise:

```suggestion
	/// We had a temporary error processing the payment. Useful if no other error codes fit 
	/// and we want to indicate that the payer may want to retry.
```